components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/jwks/JwksEndpoint.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.oauth.endpoint.jwks;
+
+
+import java.io.FileInputStream;
+import java.math.BigInteger;
+import java.security.KeyStore;
+import java.security.cert.Certificate;
+import java.security.interfaces.RSAPublicKey;
+
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.wso2.carbon.core.util.KeyStoreManager;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.utils.CarbonUtils;
+
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.MediaType;
+
+
+@Path("/jwks/{tenantDomain : (/tenantDomain)?}")
+public class JwksEndpoint {
+    private static final Log log = LogFactory.getLog(JwksEndpoint.class);
+    private static final char[] ENCODE_MAP = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_".toCharArray();
+    private static final String alg = "RS256";
+    private static final String use = "sig";
+    private static final String kid = "d0ec514a32b6f88c0abd12a2840699bdd3deba9d";
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public String jwks(@PathParam("tenantDomain") String tenantDomain) {
+        RSAPublicKey publicKey = null;
+        JSONObject jwksJson = new JSONObject();
+        try {
+            if (StringUtils.isEmpty(tenantDomain)) {
```

Comment:
Here it is not necessary to check isBlank as, if we get a white space as tenant domain it validates it as null. So StringUtils.isEmpty(tenantDomain) return true